Lib/test/test_threading.py

```diff
@@ -2143,7 +2143,9 @@ def work():
         for name in tests:
             encoded = name.encode(encoding, "replace")
             if truncate is not None:
-                expected = os.fsdecode(encoded[:truncate])
+                encoded = encoded[:truncate]
+            if sys.platform.startswith("solaris"):
+                expected = encoded.decode("utf-8")
```

Comment:
```suggestion
                expected = encoded.decode("utf-8", "surrogateescape")
```
Otherwise the test will fail on Solaris.